frontend/lib/src/components/shared/StreamlitMarkdown/styled-components.ts

```diff
@@ -154,7 +154,7 @@ export const StyledLinkIcon = styled.a(({ theme }) => ({
 
 export const StyledHeadingWithActionElements = styled.div(({ theme }) => ({
   "h1, h2, h3, h4, h5, h6, span": {
-    scrollMarginTop: theme.spacing.threeXL,
+    scrollMarginTop: theme.sizes.headerHeight,
```

Comment:
this is the height of the headerbar, which was changed [here](https://github.com/streamlit/streamlit/pull/8554/files#diff-c4906e16d3e24724e4cddf79cf7a4e37100d6d04aa6dcc3251374fcb89b2211cR151) and led to overlaying the headings. This is the better size to use